File: app/src/main/java/com/paycraft/ticketbooker/adapters/StepsPagerAdapter.java
package com.paycraft.ticketbooker.adapters;


import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.paycraft.ticketbooker.view.fragment.ConfirmationFragment;
import com.paycraft.ticketbooker.view.fragment.OTPFragment;
import com.paycraft.ticketbooker.view.fragment.RegistrationFragment;
import com.paycraft.ticketbooker.view.fragment.StationsFragment;

public class StepsPagerAdapter extends FragmentPagerAdapter {

    public StepsPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public int getCount() {
        return 4;
    }

    @Override
    public Fragment getItem(int position) {
      switch (position) {
          case 0:
          default:
           return RegistrationFragment.newInstance();
          case 1:
              return OTPFragment.newInstance();
          case 2:
              return StationsFragment.newInstance();
          case 3:
              return ConfirmationFragment.newInstance();

      }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return "Page " + position;
    }
}

File: app/src/main/java/com/paycraft/ticketbooker/adapters/StepsPagerAdapter.kt
package com.paycraft.ticketbooker.adapters


import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.paycraft.ticketbooker.view.fragment.ConfirmationFragment

import com.paycraft.ticketbooker.view.fragment.OTPFragment
import com.paycraft.ticketbooker.view.fragment.RegistrationFragment
import com.paycraft.ticketbooker.view.fragment.StationFragment

class StepsPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm,FragmentPagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getCount(): Int {
        return 4
    }

    override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> return RegistrationFragment.newInstance()
            1 -> return OTPFragment.newInstance()
            2 -> return StationFragment.newInstance()
            3 -> return ConfirmationFragment.newInstance()
            else -> return RegistrationFragment.newInstance()
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return "Page $position"
    }
}